source/common/stats/symbol_table_impl.cc

```diff
@@ -243,10 +243,16 @@ void SymbolTableImpl::incRefCount(const StatName& stat_name) {
   Thread::LockGuard lock(lock_);
   for (Symbol symbol : symbols) {
     auto decode_search = decode_map_.find(symbol);
-    ASSERT(decode_search != decode_map_.end());
 
+    ASSERT(decode_search != decode_map_.end(),
+           "Please see "
+           "https://github.com/envoyproxy/envoy/blob/master/source/docs/stats.md#"
+           "debugging-symbol-table-asserts");
     auto encode_search = encode_map_.find(decode_search->second->toStringView());
-    ASSERT(encode_search != encode_map_.end());
+    ASSERT(encode_search != encode_map_.end(),
```

Comment:
A lookup failure in the encode map but not the decode would be... doubly confusing, right? I presume these maps are kept in sync elsewhere.

This is fine, just that the second assert seems like it'd point to something more nefarious than the first.

Reply:
yeah these maps are kept in sync under lock, so I don't expect that could happen.